homeassistant/components/binary_sensor/bayesian.py

```diff
@@ -0,0 +1,173 @@
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.binary_sensor import (BinarySensorDevice,
+                                                    PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_NAME, STATE_UNKNOWN, CONF_DEVICE_CLASS)
+from homeassistant.core import callback
+from homeassistant.helpers import condition
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PROBABILITY_THRESHOLD = 'probability_threshold'
+CONF_OBSERVATIONS = 'observations'
+CONF_PRIOR = 'prior'
+
+DEFAULT_NAME = 'BayesianBinary'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME):
+    cv.string,
+    vol.Required(CONF_OBSERVATIONS):
+    vol.Schema([dict]),
+    vol.Required(CONF_PRIOR):
+    vol.Coerce(float),
+    vol.Required(CONF_PROBABILITY_THRESHOLD):
+    vol.Coerce(float),
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Threshold sensor."""
+    name = config.get(CONF_NAME)
+    observations = config.get(CONF_OBSERVATIONS)
+    prior = config.get(CONF_PRIOR)
+    probability_threshold = config.get(CONF_PROBABILITY_THRESHOLD)
+    device_class = config.get(CONF_DEVICE_CLASS)
+
+    async_add_devices([
+        BayesianBinarySensor(hass, name, prior, observations,
+                             probability_threshold, device_class)
+    ], True)
+    return True
+
+
+class BayesianBinarySensor(BinarySensorDevice):
+    """Representation of a Bayesian sensor."""
+
+    def __init__(self, hass, name, prior, observations, probability_threshold,
+                 device_class):
+        """Initialize the Bayesian sensor."""
+        self._hass = hass
+        self._name = name
+        self._observations = observations
+        self._probability_threshold = probability_threshold
+        self._device_class = device_class
+        self._deviation = False
+        self.prior = prior
+        self.probability = prior
+
+        self.current_obs = OrderedDict({})
+
+        self.entity_obs = {obs['entity_id']: obs for obs in self._observations}
+
+        self.watchers = {
+            'numeric_state': self._process_numeric_state,
+            'state': self._process_state
+        }
+
+        @callback
+        # pylint: disable=invalid-name
+        def async_threshold_sensor_state_listener(entity, old_state,
+                                                  new_state):
+            """Handle sensor state changes."""
+            if new_state.state == STATE_UNKNOWN:
+                return
+
+            entity_obs = self.entity_obs[entity]
+            platform = entity_obs['platform']
+
+            self.watchers[platform](entity_obs)
+
+            prior = self.prior
+            for obs in self.current_obs.values():
+                prior = self._update_probability(obs, prior)
```

Comment:
Good catch! It actually didn't matter (and hence didn't fail my tests) because I was treating the scenario a bit naively (`P(Event|False) = 1 - P(Event|True)`).  I thought this would be a decent first shot, but after some reflection, I preferred to allow the user to specify `P(Event|False)` and default to the above case if they do not.